secure/key-manager/src/tests.rs

```diff
@@ -146,11 +159,21 @@ impl<T: LibraInterface> LibraInterfaceTestHarness<T> {
     fn take_all_transactions(&self) -> Vec<Transaction> {
         self.submitted_transactions.borrow_mut().drain(..).collect()
     }
+
+    /// If called, will set mock_libra_timestamp to true, meaning that all libra_timestamp() calls
+    /// will return monotonically increasing values.
+    fn mock_libra_timestamp(&mut self) {
+        self.mock_libra_timestamp = true;
+    }
 }
 
 impl<T: LibraInterface> LibraInterface for LibraInterfaceTestHarness<T> {
     fn libra_timestamp(&self) -> Result<u64, Error> {
-        self.libra.libra_timestamp()
+        if self.mock_libra_timestamp {
+            Ok(self.libra_timestamp.fetch_add(1, Ordering::Relaxed))
+        } else {
+            self.libra.libra_timestamp()
+        }
     }
```

Comment:
I should have probably called this out earlier, but you've effectively reinvented libra time service here. Why do we have it twice? Also is there ever a time where we call into libra_stamp and it isn't auto-increment? I had thought of adding auto-increment (and actually did, before ultimately reverting it) -- it makes tests harder to understand especially when they fail.

Reply:
Thanks for asking this David. To be honest, this one actually confused me a little:

1. Without mocking this behaviour, I couldn't actually get the libra_timestamp() to auto increment, which caused these tests to fail. In fact, I was unable to find any code in the codebase that actually increments the timestamp? Does this actually exist?

2. My understanding is that libra time service is different to the timestamp resource (unless the timestamp resource is already being implemented/updated by an on-chain time service? Is this what happens?)

Reply:
Thanks for the quick discussion offline, @davidiw! I've updated the test to increment the on-chain libra_timestamp without having to mock it, which seems a little more obvious. The only downside is having to increment the timestamp every time we call evaluate(), however, I think that's the best we can do, so it's a little bit of a trade-off. Let me know what you think!